mm/mempool/mempool.c

```diff
@@ -46,43 +46,56 @@
  * Private Functions
  ****************************************************************************/
 
-static inline FAR sq_entry_t *mempool_remove_queue(FAR sq_queue_t *queue)
+static inline FAR dq_entry_t *
+mempool_remove_queue(FAR struct mempool_s *pool, FAR dq_queue_t *queue)
 {
-  if (!sq_empty(queue))
-    {
-      FAR sq_entry_t *entry = queue->head;
+  FAR dq_entry_t *ret = queue->tail;
```

Comment:
I think alloc the buffer from the tail of the queue will cause more cache miss. If this PR just to facilitate KASAN detection, it is better to add condition check if KASAN is enabled.

Reply:
In addition, please consider abstract link list implementation. single is faster than double queue.